internal/node/reverse_proxy/proxy.go

```diff
@@ -52,24 +58,27 @@ func (r RuleSet) AddRule(rule *node.ReverseProxyRule) error {
 	return nil
 }
 
-func (r RuleSet) Remove(name string) error {
-	if _, ok := r[name]; !ok {
+func (rs RuleSet) Remove(name string) error {
+	if _, ok := rs.rules[name]; !ok {
 		return fmt.Errorf("rule %s does not exist", name)
 	}
-	delete(r, name)
+	delete(rs.rules, name)
 	return nil
 }
 
 type RuleHandler interface {
 	Type() string
 	Match(url *url.URL) bool
 	Handler() http.Handler
+	Rank() int
 }
 
 type FileServerRule struct {
 	Matcher string
 	Path    string
-	FS      fs.FS // Optional, instead of using Path, pass in an fs.FS. Useful for embedding the Habitat frontend.
+
+	FS       fs.FS  // Optional, instead of using Path, pass in an fs.FS. Useful for embedding the Habitat frontend.
```

Comment:
should we use a `type Option` here? (like in https://pkg.go.dev/github.com/bradenaw/retry#Retry)

Reply:
Good idea - I think that can go into the upcoming refactor for proxy stuff, since there are a couple other ways this is clunky.